plugins/engine-datafusion/src/main/java/org/opensearch/datafusion/search/DatafusionQuery.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.datafusion.search;
+
+import java.util.List;
+
+/**
+ * Datafusion query class which accepts susbtrait bytes
+ */
+public class DatafusionQuery {
+    private final byte[] substraitBytes;
+
+    // List of Search executors which returns a result iterator which contains row id which can be joined in datafusion
+    private final List<SearchExecutor> searchExecutors;
+
+    public DatafusionQuery(byte[] substraitBytes, List<SearchExecutor> searchExecutors) {
+        this.substraitBytes = substraitBytes;
+        this.searchExecutors = searchExecutors;
+    }
```

Comment:
Is there a notion of prefilter phase similar to `can_match` in datafusion engine? Essentially, does the `List<SearchExecutor>` have all the shards which needs to be filtered out or already filtered list?